src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -22,11 +22,59 @@
 class YamlFileDumper extends FileDumper
 {
     /**
+     * Dump messages as tree, e.g. this
+     *   'foo.bar1' => 'test1',
+     *   'foo.bar2' => 'test2'
+     *
+     * dumps as follows:
+     *   foo:
+     *     bar1: test1
+     *     bar2: test2
+     *
+     * @var bool
+     */
+    private $asTree = true;
+
+    /**
+     * The level where you switch to inline YAML.
+     * Do not switch to inline YAML by default.
+     *
+     * @var int
+     */
+    private $inline = 999;
+
+    /**
+     * Sets asTree flag
+     * 
+     * @param bool $asTree
+     */
+    public function setDumpAsTree($asTree)
```

Comment:
FileDumper doesn't pass this options to format, so their several ways to do it:
1. Override dump() function, save options internally and reset them after format() call
2. Change FileDumper - add optional third parameter to format function and pass it in dump() function.
3. Copy-paste&edit dump function from FileDumper to YamlFileDumper (bad, very bad).
4. or leave it as setter function because if we need it dumping as tree we probably need it in whole application, not just in one case.

I think, way 2. is most suitable, but I'm not sure that it's not BC-break. Is this way ok?